lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -1861,7 +1861,9 @@ bool SROAGlobalAndAllocas(HLModule &HLM, bool bHasDbgInfo) {
           WorkList.push(*iter);
           GlobalVariable *EltGV = cast<GlobalVariable>(*iter);
           if (bHasDbgInfo) {
-            StringRef EltName = (*iter)->getName().ltrim(dbgOffset.base->getName());
+            StringRef OriginEltName = EltGV->getName();
+            StringRef OriginName = dbgOffset.base->getName();
+            StringRef EltName = OriginEltName.substr(OriginName.size());
```

Comment:
This worries me.  the basic difference between what was here before is that now the number of characters present in `OriginName` will be skipped whether they match `OriginName` or not. That seems like it might cause errors. Is that really want is wanted? If they don't match, shouldn't we fail somehow?